DEV_GUIDELINES.md

```diff
@@ -0,0 +1,135 @@
+
+
+**Reusable Components (`ui.[...]`)**
+
+Try to use the reusable components in `ui` whenever possible.
+
+If you need a custom component for your view, see whether you can wrap an existing component instead of writing your own.
+
+For example, if you need a specially styled button, try to wrap `<Button>` from `ui` instead of creating your own `<button>` element.
+
+**Component Naming**
+
+* Include the type of the component as a **suffix **of the name. For instance:
+    * `PrimaryButton`
+    * `SamplesView`
+    * `HeatmapVisualization`
+
+
+**Component Structure**
+
+If your component is simple and has no styling, a single `.jsx` file will do.
+
+If your component has styling, you should create a `.scss` file in the same folder as the `.jsx`. (Also see styling section below)
+
+As your component gets more complex, consider breaking it up into multiple components. One way is to have a [root container component and multiple presentational components](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0) that focus on specific, well-defined parts of the component (for example, for a `Table` component, you can have `<Table>`, `<Header>`, and `<Row>`.
+
+Once your component has more than a single `.jsx` and .`scss` file, you should create a separate folder to hold your component. Follow the convention below:
+
+* `Table`
+    * `index.jsx`
+    * `Table.jsx`
+    * `Table.scss`
+    * `Header.jsx`
+    * `Header.scss`
+    * `Row.jsx`
+    * `Row.scss`
+
+
+Where the `index.jsx` file is a simple redirect:
+
+`// Contents of index.jsx`
+`import Table from "./Table";`
+`export default Table;`
+
+
+**Views**
+
+Views represent a page or a section of a page in IDSeq. Views are stored in the `views` folder. 
+When designing your view, try to make as much use of the components in the `ui `and `visualizations` folders. This should be mostly plug-and-play components. 
+If you need to place a ui component in a customized place for your view (i.e. if the layout elements cannot help you), wrap it in a `div` with a proper class name, and style it in the view's css file.
+
+### REact-D3 integration 
+
+*WIP*
+D3 code should be created in plain JS (no JSX) and should be placed in separate files for React code.
+
+
+### _CSS_
+
+**File Structure**
+
+* We use one scss file per React component, which we place in the same directory as the React component.
+* If you need to create related components with common styling (like variations of a component), create a base component that holds all relevant common characterisitcs. The other components will compose this base component.  For instance, we defined `PrimaryButton` and `SecondaryButton` render a `Button` and common properties are defined in `button.scss` (even if Button does not do much more than rendering the equivalent `Button` class from `semantic-ui`).
+
+**Naming**
+
+1. **SCSS Filenames **should use the `snake_case` notation by splitting words with **underscores**
```

Comment:
Remove space between Filenames and **. Otherwise, won't reader correctly. There are other places to do that too.